pkg/tsdb/stackdriver/stackdriver.go

```diff
@@ -116,6 +116,15 @@ func (query *stackdriverQuery) isSLO() bool {
 	return query.Slo != ""
 }
 
+func isValidEmail(email string) bool {
+	if email == "" {
+		return true
+	}
+
+	re := regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
```

Comment:
Is this impressive regex copied from StackOverflow? I recall seeing one there :)

Reply:
I no more recall from where I have copied, but I didn't compiled it myself for sure.